pkg/policy/api/l4.go

```diff
@@ -156,6 +156,36 @@ type PortRule struct {
 	Rules *L7Rules `json:"rules,omitempty"`
 }
 
+// GetPortProtocols returns the Ports field of the PortRule.
+func (pd PortRule) GetPortProtocols() []PortProtocol {
+	return pd.Ports
+}
+
+// GetPortRule returns the PortRule.
+func (pd *PortRule) GetPortRule() *PortRule {
+	return pd
+}
+
+// PortDenyRule is a list of ports/protocol that should be used for deny
+// policies. This structure lacks the L7Rules since it's not supported in deny
+// policies.
+type PortDenyRule struct {
+	// Ports is a list of L4 port/protocol
+	//
+	// +optional
```

Comment:
I believe we've converted all of `+optional` to `+kubebuilder:validation:Optional`.